### PR TITLE
[MM-34351] Migrate 'components/widgets/icons' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by pangyiwei**
#### Summary

Migrate 'components/widgets/icons' and associated tests to TypeScript.

There are many components here which I separated into commits, do squash if necessary.

#### Ticket Link

Fixes [GH-17285](https://github.com/mattermost/mattermost-server/issues/17285) / [MM-34351]( https://mattermost.atlassian.net/browse/MM-34351)

#### Related Pull Requests

N/A

#### Screenshots

N/A

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by devinbinnie** [comment]
Hey @pangyiwei, thanks for the contribution! Looks like there's some lint errors currently.
Let me know when this is ready for review :)

**Comment by pangyiwei** [comment]
Hi @devinbinnie, have fixed the linting errors! Could you help with the review?

File: components/widgets/icons/mattermost_logo.tsx [hunk 1]
```diff
@@ -13,19 +13,19 @@ export default function MattermostLogo(props) {
                 x='0px'
                 y='0px'
                 viewBox='0 0 500 500'
-                style={style.background}
+                enableBackground='new 0 0 500 500'
```
Comment (devinbinnie) [inline components/widgets/icons/mattermost_logo.tsx:16]: Shouldn't this be part of the `style` attribute?
```suggestion
                style={{
                    enableBackground: 'new 0 0 500 500',
                }}
```

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @pangyiwei!

**Review by hahmadia — APPROVED** [review]

**Comment by hahmadia** [comment]
@furqanmlk 
Running E2E tests will be enough for QA this PR.

**Review by furqanmlk — APPROVED** [review]
Icon related tests passed !

image

image

**Pull request merged**